src/main/java/de/tum/cit/aet/artemis/iris/service/session/IrisExerciseChatSessionService.java

```diff
@@ -172,10 +176,18 @@ public void handleStatusUpdate(ExerciseChatJob job, PyrisChatStatusUpdateDTO sta
             var message = new IrisMessage();
             message.addContent(new IrisTextMessageContent(statusUpdate.result()));
             var savedMessage = irisMessageService.saveMessage(message, session, IrisMessageSender.LLM);
+            if (statusUpdate.tokens() != null) {
+                llmTokenUsageService.saveIrisTokenUsage(job, savedMessage, session.getExercise(), session.getUser(),
+                        session.getExercise().getCourseViaExerciseGroupOrCourseMember(), statusUpdate.tokens());
```

Comment:
I think `findByIdWithMessagesAndContents` doesn't load the exercise, let alone the course for the session

Reply:
When I tested it locally, the course and exercise ID were saved correctly in the database. Is there a case where this won't work?